test/membrane_opus/decoder_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule Membrane.Opus.Decoder.DecoderTest do
+  use ExUnit.Case, async: true
+
+  alias Membrane.Opus.Decoder
+  alias Membrane.Opus.Support.Reader
+
+  @sample_opus_packets Reader.read_packets("test/fixtures/opus_packets")
+  @sample_raw Reader.read_packets("test/fixtures/raw_packets")
+
+  test "integration" do
+    import Membrane.ParentSpec
+    import Membrane.Testing.Assertions
+    alias Membrane.Testing
+
+    elements = [
+      source: %Testing.Source{
+        output: @sample_opus_packets
+      },
+      opus: Decoder,
+      sink: Testing.Sink
+    ]
+
+    links = [link(:source) |> to(:opus) |> to(:sink)]
+
+    {:ok, pipeline} =
+      Testing.Pipeline.start_link(%Testing.Pipeline.Options{
+        elements: elements,
+        links: links
+      })
+
+    Membrane.Pipeline.play(pipeline)
+
```

Comment:
You may add assertion on state change and start_of_stream to clean up message box before important assertions and to make sure `sink_buffer` assertion won't reach the timeout while the pipeline is initializing

Reply:
`assert_element_start_of_stream` is still missing and actually more important for the timeout of buffer assertion below

Reply:
start_of_stream is triggered just at the time the first buffer comes in, so I don't see how it helps. I'm also not a fan of polluting tests with assertions whose only purpose is to exceed the timeout

Reply:
> so I don't see how it helps

The point is not to extend timeout, but to count it properly. After adding SoS assertion timeout for `assert_sink_buffer` will be counted since start_of_stream, thus it will only include processing time of a buffer, not the time needed for the first buffer to arrive.

Reply:
entire 'processing' done on that sink is sending a notification with the buffer...

Reply:
And that's exactly why you should keep those timeouts separated! You could now lower the timeout for each processed buffer, but the test will start failing because of longer wait for the first one.
If you want to limit the number of assertions, replace the recently added playback change with start of stream assertion